src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salaries = new int[names.length];
+        String[] stringDateFromArray = dateFrom.split("\\.");
+        String[] stringDateToArray = dateTo.split("\\.");
+        LocalDate dateStart = LocalDate.of(Integer.parseInt(stringDateFromArray[2]),
+                Integer.parseInt(stringDateFromArray[1]),
+                Integer.parseInt(stringDateFromArray[0]));
+        LocalDate dateEnd = LocalDate.of(Integer.parseInt(stringDateToArray[2]),
+                Integer.parseInt(stringDateToArray[1]),
+                Integer.parseInt(stringDateToArray[0]));
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                String[] currentLine = data[j].split(" ");
+                int[] currentDateInt = new int[3];
+                String[] currentDateArray = currentLine[0].split("\\.");
+                for (int k = 0; k < 3; k++) {
```

Comment:
you have to many loops it will be bad for performance

Reply:
done